### PR TITLE
Swaps to vite to build library and removes dotenv

### PR DESCRIPTION
**Pull request by douglance**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,11 +53,11 @@
     "bs58check": "^2.1.2",
     "buffer": "^5.6.0",
     "crc-32": "^1.2.0",
-    "dotenv": "^16.0.0",
     "elliptic": "6.5.4",
     "eth-eip712-util-browser": "^0.0.3",
     "hash.js": "^1.1.7",
     "js-sha3": "^0.8.0",
+    "patch-package": "^6.4.7",
```
Comment (alex-miller-0) [inline package.json:60]: What's the reason for this?

File: src/__test__/utils/getters.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import seedrandom from 'seedrandom';
 
 export const getEnv = () => {
-  if (!process.env) throw new Error('env cannot be found');
-  return process.env
+  if (!import.meta.env) throw new Error('env cannot be found');
```
Comment (alex-miller-0) [inline src/__test__/utils/getters.ts:4]: Where does `import.meta` come from?

File: package.json (hunk 1 above)
Comment (alex-miller-0) [reply package.json:60]: Ah okay. Seems like it should be in dev deps lol

Comment (alex-miller-0) [reply package.json:60]: It does get called in `postinstall` but I think that can still use devdeps right?

Comment (douglance) [reply package.json]: Sorry--missed this. It's been moved [851bf2e](https://github.com/GridPlus/gridplus-sdk/pull/439/commits/851bf2e208c855edd204037c696f211d1510146e)

**Review by alex-miller-0 — APPROVED** [review]

**Pull request merged**